tools/stats/scribe.py

```diff
@@ -40,3 +54,105 @@ def _send_to_scribe_via_http(access_token: str, logs: str) -> str:
     )
     r.raise_for_status()
     return str(r.text)
+
+
+def _rds_invoke(events: List[Dict[str, Any]]) -> Any:
+    res = aws_lambda().invoke(
+        FunctionName="rds-proxy", Payload=json.dumps(events).encode()
+    )
+    payload = str(res["Payload"].read().decode())
+    if res.get("FunctionError"):
+        raise Exception(payload)
+
+    return json.loads(payload)
+
+
+def register_rds_table(table_name: str, schema: Dict[str, str]) -> None:
+    base = {
+        "pr": "string",
+        "ref": "string",
+        "branch": "string",
+        "workflow_id": "string",
+    }
+
+    event = [
+        {
+            "create_table": {
+                "table_name": table_name,
+                "fields": {**schema, **base},
+            }
+        }
+    ]
+
+    _rds_invoke(event)
+
+
+def schema_from_sample(data: Dict[str, Any]) -> Dict[str, str]:
+    schema = {}
+    for key, value in data.items():
+        if isinstance(value, str):
+            schema[key] = "string"
+        elif isinstance(value, int):
+            schema[key] = "int"
+        else:
+            raise RuntimeError(f"Unsupported value type: {key}: {value}")
+    return schema
+
+
+Query = Dict[str, Any]
+
+
+def rds_query(queries: Union[Query, List[Query]]) -> Any:
+    if not isinstance(queries, list):
+        queries = [queries]
+
+    events = []
+    for query in queries:
+        events.append({"read": {**query}})
+
+    return _rds_invoke(events)
+
+
+def rds_saved_query(query_names: Union[str, List[str]]) -> Any:
+    if not isinstance(query_names, list):
+        query_names = [query_names]
+
+    events = []
+    for name in query_names:
+        events.append(
+            {
+                "read": {
+                    "saved_query_name": name,
+                }
+            }
+        )
+
+    return _rds_invoke(events)
+
+
+def rds_write(table_name: str, values_list: List[Dict[str, Any]]) -> None:
+    base = {
+        "pr": os.environ.get("CIRCLE_PR_NUMBER"),
+        "ref": os.environ.get("CIRCLE_SHA1"),
+        "branch": os.environ.get("CIRCLE_BRANCH"),
+        "workflow_id": os.environ.get("CIRCLE_WORKFLOW_ID"),
```

Comment:
What will happen if this function is invoked from GHA environment? Do we really want to send `None` to the RDS?

Reply:
Yeah it can handle a `null` variable just fine since they are passed through to the db which will encode it as missing